### ISSUE TITLE
[BUG] Cloud credential deletion fails intermittently in Rancher v2.7.0

### ISSUE THREAD

**Issue by stormqueen1990** [rancher/rancher#40520]
**Rancher Server Setup**
- Rancher version: v2.7.0
- Installation option (Docker install/Helm Chart): Helm chart
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE1 with Kubernetes v1.21.14 and v1.24.4
- Proxy/Cert Details: N/A

**Information about the Cluster**
- Kubernetes version: v1.21.14 and v1.24.4
- Cluster Type (Local/Downstream): local

**User Information**
- What is the role of the user logged in? Default Admin

**Describe the bug**
When trying to delete a cloud credential via the cluster management -> cloud credentials option, an intermittent error appears with the message
```
Index with name by-cloud-cred does not exist
```

I noticed this when trying to delete a credential created via Terraform, but was able to replicate the issue solely using the GUI as well. 

**To Reproduce**

_Via GUI_
* Navigate to the cloud credentials page.
* Create a new cloud credential for Digital Ocean.
* Wait a few seconds.
* Create a second cloud credential for Digital Ocean with the same API key.
* Try deleting the first credential.
* This message appears:
   ![image](https://user-images.githubusercontent.com/698465/217958044-95b2aeb3-a1a7-4886-8f1e-dc0f2ab9cab7.png)
* Keep trying to delete the credential and notice that it eventually gets deleted upon retries.

_Via Terraform_
* Use Terraform to create a cloud credential in Rancher.

    <details>
        <summary>Sample Terraform configuration</summary>

    ```terraform
    terraform {
      required_providers {
        rancher2 = {
          source  = "rancher/rancher2"
          version = "1.25.0"
        }
      }
    }
    
    variable "api_url" {}
    
    variable "token_key" {}
    
    variable "do_token" {}
    
    provider "rancher2" {
      api_url   = var.api_url
      token_key = var.token_key
      insecure  = true
      timeout   = "60s"
    }
    
    resource "rancher2_cloud_credential" "cloud_credential" {
      name = "cloud-credential"
    
      digitalocean_credential_config {
        access_token = var.do_token
      }
    }    
    ```

    </details>

* Plan & apply the creation.
* Wait a few seconds, then plan & apply a deletion.
* Error message appears:
    ```
    Error: Error removing Cloud Credential: Bad response statusCode [500]. Status [500 Internal Server Error]. Body: [baseType=error, code=ServerError, message=Index with name by-cloud-cred does not exist] from [https://my-address/v3/cloudCredentials/cattle-global-data:id]
    ```

> **Note**: I was also able to reproduce this issue with a single cloud credential, but it is intermittent. This issue appears to be more reproducible when there are more replicas of Rancher running. I scaled down my deployment to extract logs more reliably and was unable to reproduce the issue with a single deployment replica.

**Result**
Credential deletion fails intermittently.

**Expected Result**
Credential deletion goes through.

**Additional Context**
* The error message also appears in Rancher pod logs, but enabling debug logging did not show more information:
    ```
    2023/02/13 14:55:57 [ERROR] Unknown error: Index with name by-cloud-cred does not exist
    ```
* Issue appears to be related to a race condition between replicas, as it does not seem to happen when Rancher is deployed on a single replica.
* I did not test other credential types, but I believe this is not specific to the Digital Ocean type of credential.

**Comment by boris-stojnev** [comment]
I can confirm this. I wanted to open an issue in the rancher2 tf repo. :) 

My setup is slightly different, rancher is v2.7.0, v1.24.9+rke2r1, 3-node cluster. I’m creating only one s3 cloud credential through rancher2 terraform provider, and getting the same error when trying to delete. I added sleep after deleting the downstream cluster that is using that credential, but didn’t help.

This is happening from the UI as well.

**Comment by michael-goe** [comment]
Same problem here. I found out that clicking "Delete" multiple times led to successful removal for me.

**Comment by marinoborges** [comment]
Same thing here at 2.6.9

**Comment by HarrisonWAffel** [comment]
This issue seems like a duplicate of https://github.com/rancher/rancher/issues/39300 which was resolved by this PR https://github.com/rancher/rancher/pull/39622 and will be included in the 2.7.2 release (disregard the use of `2.7.1` in the PR title, this issue was resolved before the security release `2.7.1` was announced / planned)

**Comment by ElanHasson** [comment]
This also happens when you edit a credential and it results in a new one being created that can't be deleted.

I also noticed the disabled `Name` text entry is blank. The dupe that gets created doesn't have a name. Feels related.

![image](https://user-images.githubusercontent.com/234704/219978291-4bc932b9-8455-4a33-be68-5d2c8c19cf4d.png)

**Comment by snasovich** [comment]
Closing as duplicate of https://github.com/rancher/rancher/issues/39300, will be resolved in 2.7.2.
